src/CDN/CloudFront.php

```diff
@@ -152,83 +172,77 @@ public function flush($string)
     public function flushPaths(array $paths)
     {
         if (empty($paths)) {
-            throw new \RuntimeException('Unable to flush : expected at least one path');
+            throw new \RuntimeException('Unable to flush: expected at least one path');
         }
         // Normalizes paths due possible typos since all the CloudFront's
         // objects starts with a leading slash
-        $normalizedPaths = array_map(static function ($path) {
+        $normalizedPaths = array_map(static function (string $path): string {
             return '/'.ltrim($path, '/');
         }, $paths);
 
-        $invalidationBatch = [
-            'Paths' => [
-                'Quantity' => \count($normalizedPaths),
-                'Items' => $normalizedPaths,
-            ],
-            'CallerReference' => $this->getCallerReference($normalizedPaths),
-        ];
-
-        $arguments = [
-            'DistributionId' => $this->distributionId,
-        ];
-
-        // @todo: Remove the following check and the `else` block when support for aws/aws-sdk-php < 3.0 is dropped.
-        if (class_exists(Sdk::class)) {
-            // AWS v3.x.
-            $arguments += ['InvalidationBatch' => $invalidationBatch];
-        } else {
-            // AWS v2.x.
-            $arguments += $invalidationBatch;
-        }
-
         try {
-            $result = $this->getClient()->createInvalidation($arguments);
+            $result = $this->client->createInvalidation([
+                'DistributionId' => $this->distributionId,
+                'Paths' => [
+                    'Quantity' => \count($normalizedPaths),
+                    'Items' => $normalizedPaths,
+                ],
+                'CallerReference' => $this->getCallerReference($normalizedPaths),
+            ]);
 
-            if (!\in_array($status = $result->get('Status'), ['Completed', 'InProgress'], true)) {
-                throw new \RuntimeException('Unable to flush : '.$status);
+            $status = $result->get('Status');
+
+            if (false === array_search($status, self::AVAILABLE_STATUSES, true)) {
+                throw new \RuntimeException(sprintf('Unable to determine the flush status from the given response: "%s".', $status));
             }
 
             return $result->get('Id');
         } catch (CloudFrontException $ex) {
-            throw new \RuntimeException('Unable to flush : '.$ex->getMessage());
-        }
-    }
-
-    /**
-     * For testing only.
-     *
-     * @param CloudFrontClient $client
-     */
-    public function setClient($client)
-    {
-        if (!$client instanceof CloudFrontClient) {
-            @trigger_error('The '.__METHOD__.' expects a CloudFrontClient as parameter.', E_USER_DEPRECATED);
+            throw new \RuntimeException('Unable to flush: '.$ex->getMessage(), $ex->getCode(), $ex);
```

Comment:
I wouldn't copy anything from `$ex`, everything is already in `$previous`. Instead, try to bring value with information from the current context (otherwise it's not worth catching the exception).

```suggestion
            throw new \RuntimeException(sprintf('Unable to flush paths "%s". ', implode(', ', $paths), 0, $ex);
```